ee/cis/macos-13/cis-policy-queries.yml

```diff
@@ -791,6 +791,56 @@ spec:
 ---
 apiVersion: v1
 kind: policy
+spec:
+  name: CIS - Ensure Security Auditing Is Enabled
+  platforms: macOS
+  platform: darwin
+  description: |
+    macOS's audit facility, auditd, receives notifications from the kernel when certain system calls, such as open, fork, and exit, are made. These notifications are captured and written to an audit log.
+  resolution: |
+    Automated method:
+    Ask your system administrator to deploy the following script which will enable security auditing:
+    /usr/bin/sudo /bin/launchctl load -w /System/Library/LaunchDaemons/com.apple.auditd.plist
+  query: |
+    SELECT 1 WHERE 
+      -- For all files in /var/audit UID should be owned by root
+      NOT EXISTS ( SELECT 1 FROM file WHERE path LIKE '/var/audit/%' AND uid !=0 )
+      AND
+      -- For all files in /var/audit GID should be owned by root
+      NOT EXISTS ( SELECT 1 FROM file WHERE path LIKE '/var/audit/%' AND gid !=0 )
+      AND
+      -- For all files in /var/audit MODE should be 0440 ("-r--r-----")
+      NOT EXISTS ( SELECT 1 FROM file WHERE path LIKE '/var/audit/%' AND mode != "0440" )
+      AND 
+      NOT EXISTS ( select 1 from file where path LIKE ( 
+                        -- The path we are looking for is written inside /etc/security/audit_control in a line that starts with "dir:"
+                        -- Looking immediately at this and REGEX-ing the path after it (and concatenating "/%")
+                        -- Same explanation for queries below.
+                        SELECT dir FROM ( 
+                        select line, CONCAT(regex_match(line, '^dir:(.+)',1 ), '/%') AS dir FROM file_lines WHERE path = '/etc/security/audit_control' AND line LIKE "dir:%")) 
+                        AND uid !=0 ) 
+      AND 
+      NOT EXISTS ( select 1 from file where path LIKE ( 
+                        SELECT dir FROM ( 
+                        select line, CONCAT(regex_match(line, '^dir:(.+)',1 ), '/%') AS dir FROM file_lines WHERE path = '/etc/security/audit_control' AND line LIKE "dir:%" ) ) 
+                        AND gid !=0 ) 
+      AND 
+      NOT EXISTS ( select 1 from file where path LIKE ( 
+                        SELECT dir FROM ( 
+                        select line, CONCAT(regex_match(line, '^dir:(.+)',1 ), '/%') AS dir FROM file_lines WHERE path = '/etc/security/audit_control' AND line LIKE "dir:%" ) ) 
+                        AND mode != "0440" ) 
+      AND 
+      -- For /etc/security/audit_control the MODE should be 0400 ("-r--------")
+      NOT EXISTS ( select 1 from file where path = "/etc/security/audit_control" AND mode != "0400" )
+      AND 
+      -- For /var/audit the MODE should be 0700 ("drwx------")
+      NOT EXISTS ( select 1 from file where path = "/var/audit" AND mode != "0700" );
```

Comment:
Where's this part on the PDF? Cannot find it.

Reply:
I probably misinterpreted this section to include the folder itself. But I believe you are right and I will remove this.


`
As part of defense-in-depth the /etc/security/audit_control configuration and the files in /var/audit should be owned only by root with group wheel with read-only rights and no other access allowed. `